tests/test_providers.py

```diff
@@ -546,20 +547,65 @@ def test_get_base_configuration(
     )
     mocker.patch("charmcraft.providers.get_command_environment", return_value="test-env")
     mocker.patch("charmcraft.providers.get_instance_name", return_value="test-instance-name")
-    mock_buildd_base = mocker.patch("charmcraft.providers.bases.BuilddBase")
+    mock_buildd_base = mocker.patch("craft_providers.bases.ubuntu.BuilddBase")
     mock_buildd_base.compatibility_tag = "buildd-base-v0"
 
-    providers.get_base_configuration(alias=alias, instance_name="test-instance-name")
+    providers.get_base_configuration(alias=alias_ubuntu, instance_name="test-instance-name")
 
     mock_buildd_base.assert_called_with(
-        alias=alias,
+        alias=alias_ubuntu,
         environment="test-env",
         hostname="test-instance-name",
-        snaps=[bases.buildd.Snap(name="charmcraft", channel=expected_snap_channel, classic=True)],
+        snaps=[Snap(name="charmcraft", channel=expected_snap_channel, classic=True)],
         compatibility_tag="charmcraft-buildd-base-v0.0",
     )
 
 
+@pytest.mark.parametrize(
+    "platform, snap_channel, expected_snap_channel",
+    [
+        ("linux", None, None),
+        ("linux", "edge", "edge"),
+        ("darwin", "edge", "edge"),
+        # default to stable on non-linux system
+        ("darwin", None, "stable"),
+    ],
+)
+@pytest.mark.parametrize(
```

Comment:
pedantic: premature abstraction: Unless you're going to add another CentOS base in the near future, I wouldn't parameterize this.